File: py/selfDividing.py

def selfDividing(left, right):
    result = []
    for i in range(left, right+1):
        strN = str(i)
        strL = list(strN)
        invalid = False
        for n in strL:
            if int(n) == 0:
                invalid = True
            if not invalid and i % int(n) != 0:
                invalid = True
        if not invalid:
            result.append(i)
    return result
        


print(selfDividing(1, 22))
print('hi')

File: js/selfDividing.js

function selfDividing(left, right) {
    let invalid = false;
    const result = [];
    for (let i = left; i <= right; ++i) {
        const strN = i.toString();
        const strA = strN.split('');
        invalid = false;
        
        strA.forEach((c) => {
            if (c == 0) {
                invalid = true;
            }
            if (i % parseInt(c) != 0) {
                invalid = true;
            }
        });
        if (invalid != true ) {
            result.push(i);
        }
    }
    return result;
}

console.log(selfDividing(1, 22));
console.log('hi');